oak_runtime/src/lib.rs

```diff
@@ -575,9 +588,16 @@ impl Runtime {
     fn get_node_downgraded_label(&self, node_id: NodeId, initial_label: &Label) -> Label {
         // Retrieve the set of tags that the node may downgrade.
         let node_privilege = self.get_node_privilege(node_id);
-        Label {
-            // Remove all the confidentiality tags that the Node may declassify.
-            confidentiality_tags: initial_label
+
+        // Remove all the confidentiality tags that the Node may declassify.
+        // TODO(#1631): Perform this check for each sub-lattice.
+        let confidentiality_tags = if node_privilege
+            .can_declassify_confidentiality_tags
+            .contains(&top())
+        {
+            vec![]
```

Comment:
I think this is also worth a comment, it seems a pretty crucial step and we don't want our future selves to accidentally get this wrong when changing things around.

Reply:
Done.